slack_sdk/socket_mode/aiohttp/__init__.py

```diff
@@ -296,7 +299,30 @@ async def disconnect(self):
     async def send_message(self, message: str):
         if self.logger.level <= logging.DEBUG:
             self.logger.debug(f"Sending a message: {message}")
-        await self.current_session.send_str(message)
+        try:
+            await self.current_session.send_str(message)
```

Comment:
A race condition where the background job can disconnect the current session if it's not working any more. In this scenario, reconnection will be completed shortly by the background job. By acquiring the `connect_operation_lock` before retrying, this method can wait for the newly established connection.